### DISCUSSION TITLE
How to remove slint::SharedString last char in slint-ui?

### DISCUSSION THREAD

**Discussion by fluolite** [slint-ui/slint#6380] [General] [answered]
property <bool> text;

// append a ascii char
text += "a";

// remove last ascii char（）
text.pop();

**Comment by ogoffart** [comment, marked as answer]
This feature is currently missing.
You'll have to do that in native code

It's also missing on SharedString rust though
something like so should work:
```rs
let ch = the_string.chars().rev().next()?;
let newlen = the_string.len() - ch.len_utf8();
the_string = the_string[..newlen].into();
```

**Reply by fluolite** [reply]
Expect some of the data structures like javascript array(string) to support simple logic only for UI handler
